tests/integration/test_host.py

```diff
@@ -0,0 +1,136 @@
+    bundle_path = pack_bundle(from_dir=BUNDLES / "complex_provider", to=tmp_path)
+    return await adcm_client.bundles.create(source=bundle_path)
+
+
+@pytest_asyncio.fixture()
+async def hostprovider(adcm_client: ADCMClient, hostprovider_bundle: Bundle) -> HostProvider:
+    return await adcm_client.hostproviders.create(
+        bundle=hostprovider_bundle, name="Hostprovider name", description="Hostprovider description"
+    )
+
+
+async def test_host(adcm_client: ADCMClient, hostprovider: HostProvider, cluster: Cluster) -> None:
+    await _test_host_properties(adcm_client, hostprovider, cluster)
+    await _test_host_accessors(adcm_client, hostprovider, cluster)
+    await _test_pagination(adcm_client, hostprovider, cluster)
+
+
+async def _test_host_properties(adcm_client: ADCMClient, hostprovider: HostProvider, cluster: Cluster) -> None:
+    await adcm_client.hosts.create(name="test-host", description="host description", hostprovider=hostprovider)
+    await cluster.hosts.add(host=await adcm_client.hosts.get(name__eq="test-host"))
+
+    host = await adcm_client.hosts.get()
+    assert host.name == "test-host"
+    assert (await host.hostprovider).name == hostprovider.name
+    assert (await host.cluster).name == cluster.name  # pyright: ignore[reportOptionalMemberAccess]
+    assert isinstance(host.actions, ActionsAccessor)
+    assert await host.get_status() == "down"
+    assert (await host.maintenance_mode).value == "off"
+
+
+async def _test_host_accessors(adcm_client: ADCMClient, hostprovider: HostProvider, cluster: Cluster) -> None:
+    for new_host in ["host-1", "host-2", "host-3"]:
+        await adcm_client.hosts.create(name=new_host, description="host description", hostprovider=hostprovider)
+
+    host = await adcm_client.hosts.get(name__eq="host-1")
+    assert isinstance(host, Host)
+    assert host.name == "host-1"
+
+    with pytest.raises(ObjectDoesNotExistError):
+        await adcm_client.hosts.get(name__eq="fake_host")
+
+    with pytest.raises(MultipleObjectsReturnedError):
+        await adcm_client.hosts.get(name__contains="host")
+
+    assert not await adcm_client.hosts.get_or_none(name__eq="fake_host")
+    assert isinstance(await adcm_client.hosts.get_or_none(name__contains="-1"), Host)
+
+    assert len(await adcm_client.hosts.all()) == len(await adcm_client.hosts.list()) == 4
+
+    hosts_list = await adcm_client.hosts.list(query={"limit": 2, "offset": 1})
+    assert isinstance(hosts_list, list)
+    assert len(hosts_list) == 2
+
+    hosts_list = await adcm_client.hosts.list(query={"offset": 4})
+    assert isinstance(hosts_list, list)
+    assert len(hosts_list) == 0
+
+    async for h in adcm_client.hosts.iter():
+        assert isinstance(h, Host)
+        assert "host" in h.name
+
+    await cluster.hosts.add(host=await adcm_client.hosts.get(name__eq="host-1"))
+    await cluster.hosts.add(host=await adcm_client.hosts.get(name__eq="host-2"))
```

Comment:
since you're considering these cases, pls make one of them a Filter object (cluster.hosts.add(host=Filter(...)))

and them check it's added

Reply:
done